pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -36,8 +37,61 @@ def __init__(self, runtime: Runtime, version: str, assembly: str, publish: bool,
         self.logger = runtime.logger
         self.working_dir = self.runtime.working_dir
         self.fail_count_name = f'{version}-assembly.{assembly}.count'
+        self.job_run = self.runtime.get_job_run_url()
+
+    async def comment_on_assembly_pr(self, text_body):
+        """
+        Comment the link to this jenkins build on the assembly PR if it was triggered automatically
+        """
+        # Keeping in try-except so that job doesn't fail because of any error here
+        try:
+            base_repo = "openshift-eng"
+            repository = "ocp-build-data"
+            branch = self.doozer_data_gitref
+            token = os.environ.get('GITHUB_TOKEN')
+
+            pattern = r"github\.com/([^/]+)/"
+            match = re.search(pattern, self.data_path)
+
+            repo_owner = None
+            if match:
+                repo_owner = match.group(1)
+
+            api = GhApi(owner=base_repo, repo=repository, token=token)
+
+            # Head needs to have the repo name prepended for GhApi to fetch the correct one
+            head = f"{repo_owner}:{branch}"
+            # Find our assembly PR.
+            prs = api.pulls.list(head=head, state="open")
+
+            if len(prs) != 1:
+                self.logger.error(
+                    f"{len(prs)} PR(s) were found from the auto generated branch {branch}. We need only 1.")
+                return
+
+            if len(prs) == 0:
+                self.logger.error(f"No assembly PRs were found with head={head}")
+                return
+
+            pr_number = prs[0]["number"]
+
+            if self.runtime.dry_run:
+                self.logger.warning(f"[DRY RUN] Would have commented on PR {base_repo}/{repository}/pull/{pr_number} "
+                                    f"with the message: \n {text_body}")
+                return
+
+            # https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28#create-an-issue-comment
+            # PR is an issue as far as  GitHub API is concerned
+            api.issues.create_comment(issue_number=pr_number, body=text_body)
+        except Exception as e:
+            self.logger.error(f"Error commenting to PR: {e}")
 
     async def run(self):
+        if self.assembly != 'stream' and self.doozer_data_gitref is not None:
```

Comment:
`doozer_data_gitref` may not be present. If build-sync ran without it
then we would still like to try and leave a comment if there's a PR present

Reply:
added in https://github.com/openshift-eng/aos-cd-jobs/pull/3892/commits/545b6f0d09a4124236d83c23f986d1dacf4c126e